### PR TITLE
Adding Switch OSS

### PR DESCRIPTION
**Pull request by BenedicteLeo**
This commit contains changes that enable MLP OSS functionality on the Switch (joins, invites, host session, matchmaking, pubsub).

### CODE REVIEW

File: Platforms/Switch/packages.config [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Microsoft.PlayFab.PlayFabParty.Cpp.Switch-12.3.11" version="1.7.7" targetFramework="native" />
```
Comment (nassosterz-ms) [inline Platforms/Switch/packages.config:3]: I might be kate asking here, but arent Switch 12.x versions deprecated?

File: Source/OnlineSubsystemPlayFab.Build.cs [hunk 2]
```diff
@@ -238,6 +248,83 @@ public void ConfigurePlayFabDependencies(ReadOnlyTargetRules Target, ModuleRules
...
+        public bool IsPCPlatform(ReadOnlyTargetRules Target)
+        {
+            return false;
+        }
+
+        public void AddModuleDependencies(ModuleRules ThisModule)
+        {
+            ThisModule.PublicDependencyModuleNames.Add("OnlineSubsystemSwitch");
+        }
+        public void SetPlatformDefinitions(ModuleRules ThisModule)
```
Comment (nassosterz-ms) [inline Source/OnlineSubsystemPlayFab.Build.cs:262]: You may wanna leave a newline between each function definition

File: Source/OnlineSubsystemPlayFab.Build.cs [hunk 3]
```diff
@@ -238,6 +248,83 @@ public void ConfigurePlayFabDependencies(ReadOnlyTargetRules Target, ModuleRules
...
+            {
+                throw new BuildException("PlayFab precompiled dependencies were not found.");
+            }        
+
+            // Find the MLP and Party library names under the PlatformDir.
+            string[] PlatformDirectories = System.IO.Directory.GetDirectories(PlatformDir);
+            string MultiplayerDirectory = PlatformDirectories.Where(d => d.Contains("PlayFab.Multiplayer")).FirstOrDefault();
+            string PartyDirectory = PlatformDirectories.Where(d => d.Contains("PlayFab.PlayFabParty")).FirstOrDefault();
+            
+            if(String.IsNullOrWhiteSpace(MultiplayerDirectory) || String.IsNullOrWhiteSpace(PartyDirectory))
```
Comment (nassosterz-ms) [inline Source/OnlineSubsystemPlayFab.Build.cs:280]: Whitespace after if

File: Source/Private/OnlineSubsystemPlayFab.cpp [hunk 4]
```diff
@@ -212,11 +215,11 @@ const static TMap<FString, PartyDirectPeerConnectivityOptions> ConnectivityOptio
 	{"SamePlatformType", PartyDirectPeerConnectivityOptions::SamePlatformType},
 	{"DifferentPlatformType", PartyDirectPeerConnectivityOptions::DifferentPlatformType},
 	{"AnyPlatformType", PartyDirectPeerConnectivityOptions::SamePlatformType |
-			    PartyDirectPeerConnectivityOptions::DifferentPlatformType},
+						PartyDirectPeerConnectivityOptions::DifferentPlatformType},
```
Comment (nassosterz-ms) [inline Source/Private/OnlineSubsystemPlayFab.cpp:218]: Identation is off here

File: Source/Private/OnlineSubsystemPlayFab.cpp [hunk 5]
```diff
@@ -212,11 +215,11 @@ const static TMap<FString, PartyDirectPeerConnectivityOptions> ConnectivityOptio
...
 	{"SameEntityLoginProvider", PartyDirectPeerConnectivityOptions::SameEntityLoginProvider},
 	{"DifferentEntityLoginProvider", PartyDirectPeerConnectivityOptions::DifferentEntityLoginProvider},
 	{"AnyEntityLoginProvider", PartyDirectPeerConnectivityOptions::SameEntityLoginProvider |
-				   PartyDirectPeerConnectivityOptions::DifferentEntityLoginProvider},
+							   PartyDirectPeerConnectivityOptions::DifferentEntityLoginProvider},
```
Comment (nassosterz-ms) [inline Source/Private/OnlineSubsystemPlayFab.cpp:222]: Identation is off here

File: Source/Public/OnlineSubsystemPlayFab.h [hunk 6]
```diff
@@ -155,7 +158,7 @@ class ONLINESUBSYSTEMPLAYFAB_API FOnlineSubsystemPlayFab : public FOnlineSubsyst
 	int32 MaxUserCount = 8;
 	int32 MaxUsersPerDeviceCount = 1;
 	PartyDirectPeerConnectivityOptions DirectPeerConnectivityOptions = PartyDirectPeerConnectivityOptions::AnyPlatformType |
-									   PartyDirectPeerConnectivityOptions::AnyEntityLoginProvider;
+																	   PartyDirectPeerConnectivityOptions::AnyEntityLoginProvider;
```
Comment (nassosterz-ms) [inline Source/Public/OnlineSubsystemPlayFab.h:161]: Identation is off

**Review by nassosterz-ms — APPROVED** [review]

File: Source/OnlineSubsystemPlayFab.Build.cs (hunk 2 above)
Comment (BenedicteLeo) [reply Source/OnlineSubsystemPlayFab.Build.cs:262]: I will fix our internal repo, but I don't think we should block this release.

File: Platforms/Switch/packages.config (hunk 1 above)
Comment (BenedicteLeo) [reply Platforms/Switch/packages.config:3]: Unreal 4.27 supports 12.3.2 at the latest, and has unpredictable results above 12.3.2.

Comment (BenedicteLeo) [reply Platforms/Switch/packages.config:3]: I just checked Perforce to make sure.

File: Source/OnlineSubsystemPlayFab.Build.cs (hunk 3 above)
Comment (BenedicteLeo) [reply Source/OnlineSubsystemPlayFab.Build.cs:280]: Fixing on our side

File: Source/Private/OnlineSubsystemPlayFab.cpp (hunk 5 above)
Comment (BenedicteLeo) [reply Source/Private/OnlineSubsystemPlayFab.cpp:222]: Fixing on our side

File: Source/Public/OnlineSubsystemPlayFab.h (hunk 6 above)
Comment (BenedicteLeo) [reply Source/Public/OnlineSubsystemPlayFab.h:161]: Fixing on our side

File: Source/Private/OnlineSubsystemPlayFab.cpp (hunk 4 above)
Comment (BenedicteLeo) [reply Source/Private/OnlineSubsystemPlayFab.cpp:218]: Fixing on our side

**Comment by PushpadantK** [comment]
> 	"Version": 2.00,

can you please update the version here as well.

---
Refers to: OnlineSubsystemPlayFab.uplugin:4 in 3ba8ce1. [](commit_id = 3ba8ce1dcc09d8c9ff082303c71474ab370c54b8, deletion_comment = False)

File: README.md [hunk 7]
```diff
@@ -19,10 +19,10 @@ The quickstart guide for OnlineSubsystemPlayFab can be found at:
...
+|2.2.0
 
 ## Compatibility
 |Unreal Engine version|4.27|
 |-|-|
-Multiplayer SDK|v1.1.0
-Platforms|GDK, Win64+Steam
\ No newline at end of file
+Multiplayer SDK|v1.1.0+
+Platforms|GDK, Win64+Steam, Switch
```
Comment (PushpadantK) [inline README.md:28]: >Switch [](http://example.com/codeflow?start=28&length=6)

nit : Nintendo Switch

File: Platforms/Switch [hunk 8]
```diff
@@ -1 +0,0 @@
-b8aaea11329cf5e545194ae0d32a1044e0e44bf4
```
Comment (PushpadantK) [inline Platforms/Switch:1]: How is that this is getting deleted when it was not even there before?

**Review by PushpadantK — APPROVED** [review]
:shipit:

File: Platforms/Switch (hunk 8 above)
Comment (BenedicteLeo) [reply Platforms/Switch:1]: It looks like Michael added that on April 4th.

**Pull request merged**